### ISSUE TITLE
Transformer: Preload images

### ISSUE THREAD

**Issue by schlessera** [ampproject/amp-wp#4457]
## Feature description

Add a new transformer `BrowserHints` that adds browser hints as meta tags to the `<head>` node. The initial implementation should add the following hint:

- Add `preload` hint for the first X images in the page

The number X should be configurable by the user and default to 5.

Related #4213 

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* Generated HTML output from the Optimizer contains one or more resource hints that tell the browser to preload images
* Images selected for preloading should be the first X in the main post, as well as anything needed for the site header (X defaults to 5)
* Images smaller then Y pixels are ignored (Y defaults to 150)
* Images with an aspect ratio larger than Z are ignored (Z defaults to 16)
* The number of images X, the minimum size Y and the maximum aspect ratio Z are configurable by the user

## Implementation brief

* See https://github.com/ampproject/amppackager/blob/releases/transformer/transformers/preloadimage.go

## QA testing instructions

1. **Verify:** AMP optimization is enabled
2. Go to the AMP version of a page that contains multiple images
3. Use "View Source..." on the page to see the generated HTML markup
4. **Verify:** The `<head>` element contains `<link rel=preload ...>` elements pointing to the first few images

## Demo

* 

## Changelog entry

*

**Comment by westonruter** [comment]
> The number X should be configurable by the user and default to 5.

In the WordPress plugin, I think this should include:

1. Theme header image
2. Post featured image
3. Theme background image
4. Custom logo

However, it gets complicated for images that have `srcset`. In order to successfully `preload` images, we need to depend on browsers supporting `imagesrcset` and `imagesizes`. Otherwise, the browser may preload the wrong image for the viewport. 

More resources:

* https://core.trac.wordpress.org/ticket/42438#comment:12
* https://www.chromestatus.com/feature/5164259990306816
* https://docs.google.com/document/d/1wob4u9AER1kmUodqIEp78BuziSJChBl54A5ZRXPRCUI/edit#

This is only supported by Chromium-based browsers as far as I know.

**Comment by sebastianbenz** [comment]
The tricky part here is not to take away bandwidth from more critical resources (e.g. v0.js). The AMP Cache team is currently working on a (single) hero image identification heuristic. Once this has been done, we should port it to AMP Optimizer and WP.

**Comment by westonruter** [comment]
Identifying the hero image should be relatively straightforward in WordPress as it is normally the featured image that has been selected for a given post.

**Comment by sebastianbenz** [comment]
That's perfect of course, is this always going to be an image in the first viewport?

**Comment by westonruter** [comment]
Not necessarily, as a theme may also select a header image and a custom logo. But the featured image would be the most important image in the first viewport.

Here's a fairly typical scenario for images on a theme:

> ![image](https://user-images.githubusercontent.com/134745/77804923-eecdc480-703d-11ea-9fcc-9cbe5656bf22.png)

My face is the custom logo. The plants on the table are the header image. And the cloudy skyline is the post's featured image. Themes will differ in how they arrange these in a template, but they are often all visible in the first viewport.

**Comment by westonruter** [comment]
Closely related to #1895 which is for adding `preconnect` links for components used on the page. These links should only be done for components that are in the first viewport.

**Comment by schlessera** [comment]
I'm not sure this should still be included, given the work that the `PreloadHeroImage` transformer is already doing.
